### PR TITLE
Webpack 2

### PR DESCRIPTION
**Pull request by doctyper**
## Overview
- Remove support for jspm/SystemJS
- Add support for Webpack 2
- deprecate `react-wildcat-ensure` in favor of native `import()` syntax
- deprecate `react-wildcat-hot-reloader` in favor of `react-hot-loader`
- Add support for `babel-preset-env`
- Update test frameworks and setup
- Bump `react-wildcat@7.0.0`

## Synopsis
This pull request removes Wildcat support for jspm/SystemJS and introduces Webpack 2 support.

Unfortunately, we @nfl have hit a performance wall with jspm/SystemJS. We believe our compile pipeline is as optimized as we can get it, while still adhering to our [monolithic approach](https://medium.com/nfl-engineers/nfl-react-84e9cd11d384#.ir2w9q5ld). This problem has been [surfaced](https://github.com/angular/angular/issues/2531) [several](http://stackoverflow.com/questions/40971270/anyone-experience-slow-loading-times-with-angular2) times. 

The core problem is that parsing JavaScript on the client is [inherently slow](https://twitter.com/samccone/status/794387046974427136). Since SystemJS offloads module registration to the client, we lose valuable seconds to the bootstrap process. This issue is especially evident on mobile platforms, magnified by the fact that our MVP targets are mobile browsers on 3G network speeds.

Initial numbers for this migration show markedly improved time-to-first-render numbers:

### Environment Details

* OS X El Capitan 10.11.6
* Macbook Pro (Retina, 15-inch, Mid 2015)
* 2.8GHz Intel Core i7
* 16GB 1600 MHz DDR3

### Browser Details

* Chrome Version 56.0.2924.87 (64-bit)
* Incognito Mode
* Responsive Mode (iPhone 5)
* Network Throttling (Regular 3G)
* CPU Throttling (Low end device (5x slowdown))

### JSPM Timings

| Run \# | First Byte | First Render | Interactive |
|---|---|---|---|
|   1    |     --     |     6919ms     |      --     |
|   2    |     --     |     7105ms     |      --     |
|   3    |     --     |     7120ms     |      --     |
|   4    |     --     |     7150ms     |      --     |
|   5    |     --     |     7301ms     |      --     |
|        |            |                |             |
|  Avg   |     --     |     7119ms     |      --     |

### Webpack Timings

| Run \# | First Byte | First Render | Interactive |
|---|---|---|---|
|   1    |     808ms    |     4478ms     |     5286ms    |
|   2    |     918ms    |     4472ms     |     5390ms    |
|   3    |     818ms    |     4012ms     |     4830ms    |
|   4    |     902ms    |     4568ms     |     5470ms    |
|   5    |     978ms    |     4454ms     |     5432ms    |
|        |              |                |               |
|  Avg   |    884.8ms   |    4396.8ms    |    5281.6ms   |

We expect to improve on these numbers as we fine-tune our Webpack configuration.

**tldr;** Webpack 2! 🎉 🎈 👍

### CODE REVIEW

File: packages/react-wildcat-handoff/src/utils/serverRender.js [hunk 1]
```diff
@@ -1,10 +1,8 @@
-"use strict";
-
 const ReactDOM = require("react-dom/server");
 const Router = require("react-router");
 const serverContext = require("./serverContext.js");
 
-const Helmet = require("react-helmet");
+const Helmet = require("react-helmet"); // eslint-disable-line import/no-unresolved
```
Comment (jamsea) [inline packages/react-wildcat-handoff/src/utils/serverRender.js]: That's odd, why do we need to disable import/no-unresolved here?

File: shell/clean.sh [hunk 2]
```diff
@@ -19,9 +17,7 @@ for directory in packages/*; do
 
             # Remove node modules
             rm -fr node_modules;
-
-            # Remove jspm packages
-            rm -fr jspm_packages;
+            rm -f yarn.lock;
```
Comment (jamsea) [inline shell/clean.sh]: I don't think we want to remove this, this would be like removing system.config.js. Don't we want dependencies of dependencies locked?

File: shell/install-example.js [hunk 3]
```diff
@@ -12,8 +12,9 @@ cd(example);
 // Install node modules
 exec(`yarn install`);
 
-// Install jspm packages
-exec(`jspm install --log warn -y`);
+// Fix a bug where Webpack does not dedupe node_modules in linked packages
+exec(`rm -fr ./node_modules/react-helmet`);
+exec(`ln -sfv ./react-wildcat-handoff/node_modules/react-helmet ./node_modules/react-helmet`);
```
Comment (jamsea) [inline shell/install-example.js]: Ah I'm guessing why this is why we needed to add the eslint ignore unresolved line in the above file

File: example/config/webpack/base.config.js [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+} = wildcatConfig;
+
+const CACHE_DIR = ".cache";
+const CACHE_ENV_DIR = path.join(root, CACHE_DIR, NODE_ENV);
+fs.ensureDirSync(CACHE_ENV_DIR);
+
+const src = path.resolve(root, "src");
+
+const include = [
+    src // important for performance!
```
Comment (doctyper) [inline example/config/webpack/base.config.js]: Oh, this tells Babel to *only* compile files in the `src` directory. It makes sure you don't accidentally run `node_modules` through Babel.

File: packages/react-wildcat-handoff/src/utils/serverRender.js (hunk 1 above)
Comment (doctyper) [inline packages/react-wildcat-handoff/src/utils/serverRender.js]: Neither npm nor Yarn dedupe linked packages. It's not usually a big deal, but in this case Helmet needs to share the same singleton in both `react-wildcat-handoff` and `example`, in order to test and confirm the correct server-side output.

So when you run `make install`, I added logic to remove the duped `react-helmet` instance and [replace it with a symlink](https://github.com/nfl/react-wildcat/pull/194/files#diff-70d02df1f67672b7a5a7d231d92a8a46). This allows me to test the correct output, but has the side-effect of ESLint not detecting `react-helmet` because it does not follow symlinks. So the workaround here was to disable import validation.

File: shell/clean.sh (hunk 2 above)
Comment (doctyper) [inline shell/clean.sh]: `clean` no longer runs on `make install`, because doing so would wipe out Yarn's really good dependency resolution. So the meaning of `make clean` has changed to "I really want to nuke everything and re-generate all Yarn lock files".
